### ISSUE TITLE
Add note about empty place levels

### ISSUE THREAD

**Issue by dthaler** [FamilySearch/GEDCOM#495]
The discussion of `<g7:PLAC>` says:

> The principal place in which the superstructure’s subject occurred, represented as a [List](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#list) of jurisdictional entities in a sequence from the lowest to the highest jurisdiction. As with other lists, the jurisdictions are separated by commas. **Any jurisdiction’s name that is missing is still accounted for by an empty string in the list.**
> 
> The type of each jurisdiction is given in the [PLAC.FORM](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#PLAC-FORM) substructure, if present, or in the [HEAD.PLAC.FORM](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#HEAD-PLAC-FORM) structure. If neither is present, the jurisdictional types are unspecified beyond the lowest-to-highest order noted above.

The sentence "Any jurisdiction’s name that is missing is still accounted for by an empty string in the list." is somewhat counter-intuitive.  That is, the reader wonders: why would one not just remove that level from `PLAC.FORM` and omit the empty level?

For example, why have:
```
3 PLAC , Oneida, Idaho, USA
4 FORM City, County, State, Country
```
instead of:
```
3 PLAC Oneida, Idaho, USA
4 FORM County, State, Country
```
which certainly makes for smaller files?  This seems to warrant a **note** at least.

Some possible cases to consider:

* `PLAC.FORM` is absent and the `HEAD.FORM` has "City, County, State, Country", but then why not recommend adding `PLAC.FORM`?. OR,
* `PLAC.FORM` is absent and `HEAD.FORM` is absent and app defaults are "City, County, State, Country".  But relying on that being the default in all apps is error prone and certainly not implied by the spec, so using `FORM` should be recommended instead, and PR #487 explicitly adds "If neither FORM exists, the meaning of the elements are not defined in this specification beyond being names of jurisdictions of some kind, ordered from smallest to largest."  OR,
* `PLAC.FORM` and/or `HEAD.FORM` are present, but the reading app does not understand `FORM` and so ignores it.  But this is similar to them being absent, and the spec contains no mention of apps defaulting to 4 levels.

In short, the spec contains no rationale for why having an empty level at the beginning of a `PLAC` would be good behavior.   Perhaps making no recommendation either way is ok, but I think at least a note would improve clarity.  Maybe a technical FAQ discussion with several examples might also be appropriate.

Other examples to consider:

* An independent city exists in a state, and there is no county, e.g., Virginia Beach, Virginia, USA

```
2 PLAC Virginia Beach, Virginia, USA
3 FORM City, State, Country
```
vs
```
2 PLAC Virginia Beach, , Virginia, USA
3 FORM City, County, State, Country
```

* A place was recorded but it is unknown which of several jurisdictions was meant, e.g., Salem, OR or Salem, MA or...?

```
2 PLAC Salem
3 FORM City
```
vs
```
2 PLAC Salem, , ,
3 FORM City, County, State, Country
```
*

**Comment by albertemmerich** [comment]
My observation is: Many applications default to city, county, state, country and ignore any FORM (from HEAD and from the superstructure PLAC). This said
```gedcom
2 PLAC Virginia Beach, Virginia, USA
```
will show a county Virginia, and state USA in those applications. This will not happen for the often used version
```gedcom
2 PLAC Virginia Beach, , Virginia, USA
```
As long as I can describe a place by the four jurisdictions city, county, state, country, I would prefer to use the empty value for a missing / unknown jurisdiction, and not give a FORM in every call of this place.

**Comment by Norwegian-Sardines** [comment]
Personally, I think `PLAC.FORM` is an outdated concept.  The `PLAC` tag values can indicate the value is a township, county, or other indicator if necessary and with modern mapping software even with missing level values can still be used to locate a majority of places.  And as I’ve said elsewhere, a place does not have to include just city, state, country, but cemeteries, hospitals, house addresses, so a place can be located on a map.

**Comment by dthaler** [comment]
> My observation is: Many applications default to city, county, state, country and ignore any FORM (from HEAD and from the superstructure PLAC).

What do said applications do when they get PLAC payloads with a larger number of levels, such as 6?

**Comment by dthaler** [comment]
> Personally, I think `PLAC.FORM` is an outdated and little used concept. The `PLAC` tag values can indicate the value is a township, county, or other indicator if necessary and with modern mapping software even with missing level values can still be used to locate a majority of places. And as I’ve said elsewhere, a place does not have to include just city, state, country, but cemeteries, hospitals, house addresses, so a place can be located on a map.

I suspect `PLAC.FORM` was added to provide labels for use in a user interface, next to the values of the levels.
So if you have a `PLAC` payload with, say, 6 labels, then a UI that supports at least that many separate fields could get the labels from `FORM`.  My own app does not use such labels, but some apps may want to have such a UI.

**Comment by tychonievich** [comment]
Discussed in steering committee 2 JULY 2024

- We appreciate the discussion so far. There seems to be plenty of content to put in a technical FAQ or a note in the spec.
- There appears to be some fact finding about various current applications that would help us write the best description. In particular,
    - What applications (if any) truncate to 4 PLAC levels?
    - What applications (if any) assume 4 PLAC levels but also support more or fewer? How do they do so?
    - How common are applications that use the PLAC.FORM in their UI?

**Comment by fisharebest** [comment]
FWIW, my own application ignores empty place levels.

Firstly, when navigating place hierarchies and maps, it becomes necessary to create dummy `(unknown)` labels for the missing levels.
This leads to a confusing UI.

Secondly, like `NULL` in databases, missing levels are not equal.  e.g. these two places are different, because the missing level could be different.

* `Newport, , England`
* `Newport, , England`

For the same reason, I also ignore `PLAC.FORM`.

**Comment by albertemmerich** [comment]
The way of describing places (including buildings, cemetaries, churches, parishes etc) is outdated in GEDCOM. There is great demand to switch to a structure of hierarchical objects. GEDCOM 7.0 and its PLAC, PLAC.FORM structure fails to fulfill the requirements:

- places may have different names in different languages, however the place itself is the same
- places may have different names dependant on the date
- places often have changing hierarchical structures of jurisdictions

All of these conditions will result in varying PLAC payloads for the same place object, and GEDCOM 7.0 does not provide a way to show that these payloads describe the same place. A workaround may be to use EXID ans EXID.TYPE to point to an external database like the Historic Gazetteer (gov.genealogy.net). Doing this, the hierarchy of jurisdictions is no longer needed as every object in describes uniquely the place, providing the type of the object, the time dependant jurisdictions of next levels as well as the name in different languages, at differnet times and so on.

Based on these ideas the German GEDCOM-L group has defined an extension of "location records", using one record for every place object and providing all the data known for the object. In GEDCOM 5.5.1 and in GEDCOM 7.0 we cite these location records by

```gedcom
2 PLAC <place payload>
3 _LOC <XREF:_LOC>
```

By this we still have the structure in GEDCOM files which will be imported by other applications not using the _LOC records, but the place payload may vary for the same place object.

My application uses these _LOC records and the user may identify different PLAC payloads as describing the same object. The payloads can be kept as is, only identifying the object by assigning the <XREF:_LOC>. The application provides the interpretation of PLAC.FORM, too, not restricted to city objects, but including builings, cemetaries, churches and other objects. The observation is, that users use the identifying process of different place payloads merging them in one place object, but do not use the PLAC.FORM structures provided by the application.

As most applications do not interpret the FORM data, and by this the transfer of place data in between applications is based on the PLAC payload only, we should not modify any elements in the GEDCOM 7.x spec for places which would require modifications of the applications.

I propose to switch to an location record system in next major GEDCOM version, and do this as soon as possible.

**Comment by dthaler** [comment]
Discussion in GEDCOM Steering Committee 8 OCT 2024
* It is unclear how to represent the case where a source record indicates a place that is unclear what jurisdiction level is meant.  E.g., is "New York" a state or a city (there are many examples of such places where the city and a larger jurisdiction such as a county, have the same name). A `NOTE` could perhaps be used in such a case.  This is a topic for the Technical FAQ to address.

**Comment by dthaler** [comment]
Discussion 14 JAN 2025:
* There are also cases where the same name appears in multiple places "Salem" but in which state?  Or American place names that are the same as place names in Europe.  That would be good to add to a FAQ.
* We should add examples for the different possibilities.
* Another possibility might be using Unknown as a jurisdictional level
```
   2 PLAC New York
   3 FORM Unknown
```
where "Unknown" is just a word that could have been "Undefined" or "?" or whatever else in their language of choice.
* An example for a place in multiple jurisdictions but where the jurisdictional level is known:
```
   2 PLAC Salem
   3 FORM City
```
* Example for a place in multiple jurisdictions where the level name is not known (York, Pennsylvania is both a city and a county, vs York, England is both a city and an abbreviation for the county of Yorkshire):
```
   2 PLAC York
   3 FORM ?
```
* City, municipality, county, and state can all have the same name in places in Germany so there can be more than 2 levels of ambiguity in level in many cases. (Oldenburg: many cities, and one of them was capital of the former state Olden burg)
* Is the following legal:
```
2 PLAC Paris, either Indiana or Maine, USA
3 FORM City, State, Country
```

**Comment by Norwegian-Sardines** [comment]
I think, as I have written previously, that when we redesign the `PLAC` structure that we need to have two concepts of “place” as it relates to a fact/event.  1) The place as written in the Source Artifact, 2) The place identity as interpreted (and possibly resolved) by the researcher!  This concept would then eliminate the need for an “unknown” `PLAC`.`FORM` identifier.  If the Source Artifact does not present enough evidence to conclude a specific “place identity” then a `PLAC`.`NOTE` should be used to present any arguments for  a potential resolution!

**Comment by albertemmerich** [comment]
GOV-type-list: I forwarded a request to the technical team for GOV to provide a multi-language downloadable list. So far about 50% of the GOV types are defined in English, too, all of them have a German name. Other languages are supported, too, however the percentage of already given types in those languages is smaller.

All of the problems discussed so far are resolved by the GEDCOM-L _LOC model:

- you do not know to which jurisdiction the place should be assigned? No problem, the _LOC.TYPE in our model is {0:M}. So you put no type to document the status. M you need if the type was modified from time to time!
- you do not know, to which state the place Paris belongs to, but you know, it is in the US? No problem, you put the hierarchical relation _LOC._LOC directly to the US.

HEAD.PLAC.FORM: That does not help if you transfer the data to another program merging them with data written using another content of HEAD.PLAC.FORM. My first application I used had 9 elements - and failed to put them in the lowest to highest level order: I still have place payloads with these 9 comma separated parts, as the next applications I used did not care about HEAD.PLAC.FORM and PLAC.FORM. All applications took the payload of PLAC as it was - no modifications to fit to other PLAC.FORMS or no putting the HEAD.PLAC.FORM under the PLAC calls as PLAC.FORM.

By this experience I do not like the structure of comma-separated place payloads at all. As the hierarchy of jurisdiction is varying from time to time, you have to reset all payloads which belong to the highest level of jurisdictions which just has got a new structure. Moreover this payload is much too long to give good results in reports. The worst: the original name of the place taken from the source cannot be represented in a way which will survive the transfer of data to another program. By the GEDCOM-L solution you can put the name of the place as payload in PLAC just as given by the source (or as used at the time when the event happened). You do not need any comma at all, if you do not want to use it. Every _LOC record may be cited by as many PLAC calls with different payloads as you need for your data! I have individuals who lived in one and the same place - but the records show three completely different names (and even more if you take the names in other languanges, too!).

The GEDCOM-L solution offers identification by assigning the PLAC to an _LOC record, the _LOC record describing all you know about the place: its geographical coordinates, its names over the time, its next level jurisdictions ( {0:M} : 0 if you do not know, > 1 if you know varying relations in different times) and assigning the GOV-ID to uniquely define the object. Originally we did that by the extension tag _GOV, but by now using GEDCOM 7.0 we have EXID and EXID.TYPE allowing to identify the places in different external systems.

If you are not shure which identification would be the correct one, you create a _LOC object just describing what you know about the place. This will not have coordinates neither a correct relation to next level jurisdiction, but may be to the next one like a state or a country. If you are successful to later identify it you merge the object with the exactly defined one...

There ist no limitation for the places to describe: You can do it for graves as well as for houses or all farms in a village: Within this structure you are able to document the history of the farms AND their families living their in one integrated database. You can enter in a structured way any events as selling the farm, a big fire destroying the farm, a reconstruction or building new houses on the farm. And their will be no problem to transfer this by GEDCOM-L _LOC records if sending and receiving application have implemented the model.

This is what I have been doing in my own researches since a long time now.

I am open for other solutions - if they are able to transfer these data, too. We should try to improve the GEDCOM-L model, but not go backwards loosing features which already can be used!

**Comment by Norwegian-Sardines** [comment]
The location ( _LOC) database and hierarchy is a great start as a lookup source for information about places, but should not be included in the GEDCOM specification.  Assigning a “place” to a _LOC (a GOV-ID) record or some other place authority is fine but the hierarchy of places is better suited for an information repository than inclusion in a GEDCOM transmission!

> By this experience I do not like the structure of comma-separated place payloads at all. As the hierarchy of jurisdiction is varying from time to time, you have to reset all payloads which belong to the highest level of jurisdictions which just has got a new structure. Moreover this payload is much too long to give good results in reports. The worst: the original name of the place taken from the source cannot be represented in a way which will survive the transfer of data to another program.

There is a lot to unpack in this statement, some of which I’ve already addressed in other entries and proposals.

Since Albert is on the steering committee I’m sure none of those ideas have been discussed.  I’m going on holiday for the next few months so I will not have much time to replay to this thread!

**Comment by albertemmerich** [comment]
> Assigning a “place” to a _LOC (a GOV-ID) record or some other place authority is fine but the hierarchy of places is better suited for an information repository than inclusion in a GEDCOM transmission!

Same you can say for the information of individuals. I do not see why we should handle places in another way than individuals. You can put individuals into an online repository, too. And then only cite them in your GEDCOM transmission. This is not the way I think the research will go.

We have no online repository for places which is going to be complete under the level of villages. And even worse: we do not find one which shows the history of places, too. We have a lot of historical data in GOV - but not worldwide. We have systems more complete, but not historical. We have different systems which can or cannot be accessed by an application to download the detailed information. And every repository has its own way to access it - very bad for applications, which often only give online access to one repository, if any. You will find a lot of churches, cemeteries, anf other important spots, but not on the level of farms and other units on the level of buildings. In Germany we know books about the history of farms in a village, or about a single farm. And if I have ancestors living there (I do), I am interested to include information about that in my genealogy!

My statement is not in the function as member of steering committee, it is to to get the discussion for places to be more detailed and find better solutions than we already have "in the wild". GEDCOM standard is on the level of the 1990 for places, and there is no application which really follows the ideas implemented by PLAC and PLAC.FORM. This structure would bring us a mess of comma separated "names" including the often varying jurisdiction hierarchy. And you see how difficult it is to handle this outdated strategy by looking at Family Search location repository - often failing in representing the correct hierarchy by mixing elements in the jurisdiction hierarchy of different time epochs. You should not try to represent the complete history for jurisdictions for every place in an area again and again! This was the main reason why GEDCOM-L implemented a hierarchical system, where you can cite the next level. And describe this level (say: a county) only once in the system.

There are several isolated solutions in the wild, which in many cases cannot be exchanged via GEDCOM without using extensions. Exchanging data and merging data of different researchers is very time consuming - and one reason is the bad situation on the place side!

Steering committee has not the function of a PLACE team, and to create new solutions.

**Comment by mother10** [comment]
I agree with @albertemmerich  about GEDCOM-L. Thats why I tried to implement that in some way in that PR I did. Now that might not be complete, and it might need some changes or better wordings, but as it was not done somewhere else yet, I decided to give that a go. I really like the idea of describing a location in parts, each part being a level in a location hierarchie.

Also I very much agree with the fact GEDCOM describes things the 1990 way. At that time people just started with computers and therefor did their best to "invent" how things should be done, and what methods should be used.
I have reason to say so, as I come from that period and before, fortunately on a place where there was already a lot of structure used, and a lot of methods "invented" that proved their value, and were called "bright new" later while we already had used that for many years.

But we live in another time and we have to move forward, as standing still is going backwards.
Unfortunately we are a bit "stuck" with GEDCOMs full of these old ways of doing.

And we have to decide if we keep going to add "small" changes to GEDCOM, or take the step to create something modern that will serve for many years to come. 
It would be great if we could add/change some things now, that can also be used in a more modern version of GEDCOM. Like what has happened to NAME and is now being created as SPLAC for places.

What I said about adding a TYPE to an SPLAC did not mean I wanted GEDCOM to include some whole databases with types. But I ment to implement a way to describe what database was used, for instance by adding that GOVTYPE, or GEOTYPE, together with a number (from that specific database) that denoted the type of the place, as described by that GOV-list or Geonames, or whatever.
With that a user of certain software can decide to add the information that belongs to it as Note's or something, or just point to the correct place so another user, if he wished, can read it from there.
Because, as already said we dont want GEDCOM's to be too large.

Although I have a remark on that too: filelenghth was a problem in 1990 yes, as disks were expensive. But I wonder if thats still true today, with all possibilities we have locally and in the cloud.
Of course we dont want to copy whole books, or typelists etc in a GEDCOM, as when combining GEDCOMs that info might be double in the resulting GEDCOM.
What I thought more about when intruducing the SPLAC.TYPE (like in GEDCOM-L) was a kind of "TYPE enum-list", not in GEDCOM itself, but outside, so independant. Same as GEDCOM-L does.

I disagree with Norwegian-Sardines about the fact SPLAC should not be in a hierarchy of places. Thats why I kept that when adding my solution. Having a hierarchy makes things very well structured. And adds a lot of flexibility. That hierarchy already excists from the start of 2020, as an addendum/extension, and I think it proved valuable.
